File: GrooveOn.Models/ResponseObjects/PaymentResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class PaymentResponse
    {
        public int Id { get; set; }

        public int SubscriptionId { get; set; }
        public int? SubscriptionPlanId { get; set; }
        public string? SubscriptionPlanName { get; set; }

        public string PaymentStatus { get; set; } = string.Empty;

        public string? StripePaymentIntentId { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? PaidAt { get; set; }

        public string? FailureReason { get; set; }

        public int? UserId { get; set; }
        public string? Username { get; set; }
    }
}

File: UI/grooveon_mobile/lib/models/payment_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_response.g.dart';

@JsonSerializable()
class PaymentResponse {
  final int id;
  final int subscriptionId;
  final String paymentStatus;
  final String? stripePaymentIntentId;
  final DateTime createdAt;
  final DateTime? paidAt;
  final String? failureReason;
  final String? paymentMethod;
  final double paymentAmount;
  final DateTime? paymentDate;

  final int? userId;
  final String? username;
  final int? subscriptionPlanId;
  final String? subscriptionPlanName;

  PaymentResponse({
    required this.id,
    required this.subscriptionId,
    required this.paymentStatus,
    this.stripePaymentIntentId,
    required this.createdAt,
    this.paidAt,
    this.failureReason,
    this.paymentMethod,
    required this.paymentAmount,
    this.paymentDate,
    this.userId,
    this.username,
    this.subscriptionPlanId,
    this.subscriptionPlanName,
  });

  factory PaymentResponse.fromJson(Map<String, dynamic> json) =>
      _$PaymentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentResponseToJson(this);
}